packages/content/src/enrichments.ts

```diff
@@ -0,0 +1,443 @@
+    newContent.updatedDateSource = "server.editingInfo.lastEditDate";
+  }
+
+  // publishedDate & publishedDateSource:
+  // publishedDate is already set to item.created, we will override that if we have pubDate or createdDate in metadata
+  const pubDate = parseISODateString(
+    getValueFromMetadata(newContent, "pubDate")
+  );
+  const createDate = parseISODateString(
+    getValueFromMetadata(newContent, "createDate")
+  );
+  newContent.publishedDatePrecision = DatePrecision.Day;
+  if (pubDate) {
+    newContent.publishedDate = pubDate.date;
+    newContent.publishedDatePrecision = pubDate.precision;
+    newContent.publishedDateSource = getMetadataPath("pubDate");
+  } else if (createDate) {
+    newContent.publishedDate = createDate.date;
+    newContent.publishedDatePrecision = createDate.precision;
+    newContent.publishedDateSource = getMetadataPath("createDate");
+  }
+
+  return newContent;
+}
+
+const fetchGroupIds = (
+  content: IHubContent,
+  requestOptions?: IHubRequestOptions
+): Promise<string[]> => {
+  return getItemGroups(content.id, requestOptions).then(response => {
+    const { admin, member, other } = response;
+    return [...admin, ...member, ...other].map(group => group.id);
+  });
+};
+
+const fetchMetadata = (
+  content: IHubContent,
+  requestOptions?: IHubRequestOptions
+) => getContentMetadata(content.id, requestOptions);
+
+const fetchOwnerOrgId = (
+  content: IHubContent,
+  requestOptions?: IHubRequestOptions
+): Promise<string> => {
+  const options: IGetUserOptions = {
+    username: content.owner,
+    ...requestOptions
+  };
+  return getUser(options).then(user => user.orgId);
+};
+
+const fetchContentData = (
+  content: IHubContent,
+  requestOptions?: IHubRequestOptions
+) => {
+  return getItemData(content.id, requestOptions);
+};
+
+interface IEnrichmentRequests {
+  [key: string]: (
+    content: IHubContent,
+    requestOptions?: IHubRequestOptions
+  ) => Promise<unknown>;
+}
+
+const enrichmentRequests: IEnrichmentRequests = {
+  // portal only
+  groupIds: fetchGroupIds,
+  metadata: fetchMetadata,
+  // both portal and hub
+  data: fetchContentData,
+  orgId: fetchOwnerOrgId
+};
+
+// TODO: use family instead
+const shouldFetchData = (content: IHubContent) => {
+  // TODO: we probably want to fetch data by default for other types of data
+  return !content.data && includes(["template", "solution"], content.hubType);
+};
+
+const isHubCreatedContent = (content: IHubContent) => {
+  const hubTypeKeywords = ["Enterprise Sites", "ArcGIS Hub"];
+  const contentTypeKeywords = content.typeKeywords || [];
+  // currently Hub only creates web maps, so
+  // we may want to remove or modify this type check later
+  return (
+    content.type === "Web Map" &&
+    contentTypeKeywords.some(
+      typeKeyword => hubTypeKeywords.indexOf(typeKeyword) > -1
+    )
+  );
+};
+
+const shouldFetchOrgId = (content: IHubContent) => {
+  return !content.orgId && isHubCreatedContent(content);
```

Comment:
Maybe we don't need to expose `fetchOwnerOrgId` if this handles detecting content coming from composer?

Reply:
this does not detect content coming from composer.